lib/daru/category.rb

```diff
@@ -918,10 +918,20 @@ def modify_category_at pos, category
       end
       old_category = category_from_position pos
       @array[pos] = int_from_cat category
-      @cat_hash[old_category].delete pos
+      @cat_hash[old_category].delete pos unless old_category.nil?
       @cat_hash[category] << pos
     end
 
+    def set index, val
+      categories = @cat_hash.keys
+      add_category(val)
+      return self.[]= index, val if @index.include? index
+      @index = @index.add(index)
+      @array << @array.max + 1 unless categories.size == @cat_hash.keys.size
+      @cat_hash.store(val,[index])
```

Comment:
Here's a simple way this can be implemented:

1. Raise exception is category is new else continue.
2. Make changes to `@cat_hash` to add new entry.
3. Make changes to `@array` for new index.
That's it!